webhookReload.py

```diff
@@ -0,0 +1,60 @@
+import sys
+import signal
+import subprocess
+
+from dotenv import dotenv_values
+from flask import Flask, request
+
+
+# used during initialization
+def update_local_repo():
+    # pull latest commit
+    subprocess.call(["git", "remote", "add", "origin", f"git@github.com:{user}/{repo}"], stdout=subprocess.PIPE)
+    subprocess.call(["git", "fetch", "origin"], stdout=subprocess.PIPE)
+    subprocess.call(["git", "switch", branch], stdout=subprocess.PIPE)
+    subprocess.call(["git", "reset", "--hard"], stdout=subprocess.PIPE)
+
+
+def start_bot() -> subprocess.Popen:
+    return subprocess.Popen(args=["python", "main.py"], executable=sys.executable)
+
+
+def reset_bot(process) -> subprocess.Popen:
+    print(f"new commit on {branch}")
+    process.send_signal(signal.SIGTERM)
+    process.wait()
+
+    # Fetch the commit
+    subprocess.call(["git", "pull"], stdout=subprocess.PIPE)
+
+    return start_bot()
+
+
+# webhook handling code
+app = Flask(__name__)
```

Comment:
I am really not sure why this line is located here or what it does but that's probably due to inexperience.